packages/vite/src/node/env.ts

```diff
@@ -35,35 +35,33 @@ export function loadEnv(
     }
   }
 
-  for (const file of envFiles) {
+  const parsed: Record<string, any> = envFiles.reduce((prev, file) => {
     const path = lookupFile(envDir, [file], { pathOnly: true, rootDir: envDir })
-    if (path) {
-      const parsed = dotenv.parse(fs.readFileSync(path), {
-        debug: process.env.DEBUG?.includes('vite:dotenv') || undefined
-      })
+    if (!path) return prev
 
-      // let environment variables use each other
-      dotenvExpand({
-        parsed,
-        // prevent process.env mutation
-        ignoreProcessEnv: true
-      } as any)
+    const output = dotenv.parse(fs.readFileSync(path), {
+      debug: process.env.DEBUG?.includes('vite:dotenv') || undefined
+    })
+    return { ...prev, ...output }
+  }, {})
```

Comment:
[nit] I think using `Array::flatMap` + `Object::fromEntries` instead of `Array::reduce` is more easier to understand.

Reply:
I don't know how to use `Array.flatMap` + `Object.fromEntries` instead of `Array.reduce` here. Can you help me?

Also, I think using one method is easier to understand than multiple methods.

Reply:
I was thinking of something like this.
```js
const parsed = Object.fromEntries(envFiles.flatMap(file => {
  const path = lookupFile(envDir, [file], { pathOnly: true, rootDir: envDir })
  if (!path) return []
  return Object.entries(dotenv.parse(fs.readFileSync(path), {
    debug: process.env.DEBUG?.includes('vite:dotenv') || undefined
  }))
}))
```